src/Producer.php

```diff
@@ -48,84 +41,83 @@ class Producer
      *
      * @var bool
      */
-    private $autoDelete;
+    protected $autoDelete;
 
     /**
      * Non-persistent (1) or persistent (2).
      *
      * @var [type]
      */
-    private $deliveryMode;
+    protected $deliveryMode;
 
     /**
-     * The arguments that should be added to the `queue_declare` statement for dead lettering.
-     *
-     * @var array
+     * @param Vinelab\Bowler\Connection $connection
      */
-    private $arguments = [];
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
 
     /**
-     * @param Vinelab\Bowler\Connection $connection
-     * @param string                    $exchangeName
-     * @param string                    $exchangeType
-     * @param string                    $routingKey
-     * @param bool                      $passive
-     * @param bool                      $durable
-     * @param bool                      $autoDelete
-     * @param int                       $deliveryMode
+     * Setup Producer.
+     *
+     * @param string $exchangeName
+     * @param string $exchangeType
+     * @param bool   $passive
+     * @param bool   $durable
+     * @param bool   $autoDelete
+     * @param int    $deliveryMode
      */
-    public function __construct(Connection $connection, $exchangeName, $exchangeType = 'fanout', $routingKey = null, $passive = false, $durable = true, $autoDelete = false, $deliveryMode = 2)
+    public function setup($exchangeName, $exchangeType = 'fanout', $passive = false, $durable = true, $autoDelete = false, $deliveryMode = 2)
     {
-        $this->connection = $connection;
         $this->exchangeName = $exchangeName;
         $this->exchangeType = $exchangeType;
-        $this->routingKey = $routingKey;
         $this->passive = $passive;
         $this->durable = $durable;
         $this->autoDelete = $autoDelete;
         $this->deliveryMode = $deliveryMode;
     }
 
     /**
-     * Publish a message to a specified exchange.
+     * Send a message to a specified exchange.
      *
      * @param string $data
+     * @param string $routingKey    The routing key used by the exchange to route messages to bounded queues.
      */
-    public function publish($data)
+    public function send($data = null, $routingKey = null)
     {
         $channel = $this->connection->getChannel();
 
         try {
             $channel->exchange_declare($this->exchangeName, $this->exchangeType, $this->passive, $this->durable, $this->autoDelete);
         } catch (\Exception $e) {
-            throw new DeclarationMismatchException($e->getMessage(), $e->getCode(), $e->getFile(), $e->getLine(), $e->getTrace(), $e->getPrevious(), $e->getTraceAsString(), $this->compileParameters(), $this->arguments
-                    );
+            app(BowlerExceptionHandler::class)->handleServerException($e, $this->compileParameters());
         }
 
         $msg = new AMQPMessage($data, ['delivery_mode' => $this->deliveryMode]);
```

Comment:
Will this throw an exception if `$data` is **null**? Or what will happen in that case?